src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -308,22 +309,41 @@ private void handleInvalidFileError(
             null
         );
 
-        ErrorMapping
-            .getFor(cause.getClass())
-            .ifPresent(errorCode -> {
-                try {
-                    sendErrorMessageToQueue(zipFilename, containerName, eventId, errorCode, cause);
-                } catch (Exception exc) {
-                    log.error(
-                        "Error sending notification to the queue."
-                            + "File name: " + zipFilename + " "
-                            + "Container: " + containerName,
-                        exc
-                    );
-                }
-            });
+        Optionals.ifPresentOrElse(
+            ErrorMapping.getFor(cause.getClass()),
+            (errorCode) -> trySendErrorMessageAndMoveFileToRejectedContainer(
+                zipFilename,
+                containerName,
+                cause, eventId,
+                leaseId,
+                errorCode
+            ),
+            () -> {
+                throw new ConfigurationException("Error code mapping not found for " + cause.getClass().getName());
+            }
+        );
+    }
 
-        blobManager.tryMoveFileToRejectedContainer(zipFilename, containerName, leaseId);
+    private void trySendErrorMessageAndMoveFileToRejectedContainer(
+        String zipFilename,
+        String containerName,
+        Exception cause,
+        Long eventId,
+        String leaseId,
+        ErrorCode errorCode
+    ) {
+        try {
+            sendErrorMessageToQueue(zipFilename, containerName, eventId, errorCode, cause);
+            blobManager.tryMoveFileToRejectedContainer(zipFilename, containerName, leaseId);
+        } catch (Exception exc) {
+            log.error(
+                "Error sending notification to the queue."
+                    + "File name: " + zipFilename + " "
+                    + "Container: " + containerName,
+                exc
+            );
+            throw exc;
```

Comment:
I missed this change.

We should either not catch this exception at all or throw a new exception with our custom message. We can't log an error and then throw an exception, because that will result in two error log entries in App Insights, representing the same thing.